client/src/JobPaginator.jsx

```diff
@@ -1,326 +1,71 @@
-                            ' go back to the main menu and hit the "build" button' +
-                            " for this job before publishing it.";
-                        alert(msg);
-                        return;
-                    }
-                    msg =
-                        "Are you sure you wish to publish this job? You will not be able" +
-                        " to make any further edits.";
-                    if (!window.confirm(msg)) {
-                        return;
-                    }
-                }
-                this.ajax.ajax({
-                    type: "PUT",
-                    url: this.ajax.getJobUrl(this.state.category, this.state.job, ""),
-                    contentType: "application/json",
-                    data: JSON.stringify({ order: this.state.order, published: publish }),
-                    success: function () {
-                        alert("Success!");
-                        window.location.assign("/paginate"); // TODO: Route better?
-                        dispatchEvent(new Event("Prep.saved"));
-                    },
-                    error: function () {
-                        alert("Unable to save!");
-                    },
-                });
-            }.bind(this)
-        );
-    }
-
-    toggleZoom() {
-        this.newState = this.state;
-        this.newState.zoom = !this.newState.zoom;
-        this.setState(this.newState);
-    }
-
-    componentDidMount() {
-        this.loadJob(this.props.category, this.props.job);
-    }
-
-    render() {
-        var preview = this.state.zoom ? (
-            <PaginatorZoomy img={this.getImageUrl(this.state.currentPage, "large")} />
-        ) : (
-            <PaginatorPreview img={this.getImageUrl(this.state.currentPage, "medium")} />
-        );
-        return (
-            <div className={this.state.active ? "" : "hidden"} id="paginator">
-                <div className="row">
-                    <div className="six col">{preview}</div>
-                    <div className="six col">
-                        <p>
-                            {this.state.category} &gt; {this.state.job}
-                        </p>
-                        <PaginatorControls paginator={this} />
-                        <PaginatorList paginator={this} pageCount={this.state.order.length} />
-                    </div>
+import useJobPaginator from "./useJobPaginator";
+
+const JobPaginator = ({ initialCategory, initialJob }) => {
+    const {
+        state: { currentPage, zoom, order, category, job },
+        action: {
+            setLabel,
+            getLabel,
+            getMagicLabel,
+            setPage,
+            nextPage,
+            prevPage,
+            deletePage,
+            save,
+            useEffect,
+            autonumberFollowingPages,
+            getImageUrl,
+            setZoom,
+        },
+    } = useJobPaginator(initialCategory, initialJob);
+
+    useEffect();
```

Comment:
What is the purpose of an empty call to useEffect()? I haven't seen that before (but I'm fairly new to this).

Reply:
I think I'll just move loadJob into useEffect, to make this more explicit.

Reply:
Moved it.